File: src/java/eu/sig/training/ch04/v4/Transfer.java
package eu.sig.training.ch04.v4;

import eu.sig.training.ch04.Money;
import eu.sig.training.ch04.v3.CheckingAccount;

public class Transfer {
    CheckingAccount counterAccount;

    @SuppressWarnings("unused")
    public Transfer(Account acct1, CheckingAccount acct2, Money m) {}

    public CheckingAccount getCounterAccount() {
        return this.counterAccount;
    }

}

File: src/csharp/eu/sig/training/ch04/v4/Transfer.cs
using eu.sig.training.ch04.v3;

namespace eu.sig.training.ch04.v4
{

    public class Transfer
    {
        public CheckingAccount CounterAccount { get; set; }

        public Transfer(Account acct1, CheckingAccount acct2, Money m)
        {
        }
    }
}